odoo/fields.py

```diff
@@ -1853,16 +1853,22 @@ def write(self, records, value):
             for term in new_terms:
                 text2terms[self.get_text_content(term)].append(term)
 
+            is_text = getattr(self.translate, 'is_text', lambda term: True)
```

Comment:
if `is_text` is `Fasle` you can bypass the whole `typo fixing` logic (also the `text2terms` computed before)
Using `if old_is_text or not closest_is_text:` to pypass is logicially correct but to complicated